coffea/nanoevents/methods/vector.py

```diff
@@ -417,6 +598,75 @@ def delta_r(self, other):
         """
         return numpy.sqrt(self.delta_r2(other))
 
+    @awkward.mixin_class_method(numpy.negative)
+    def negative(self):
+        """Returns the negative of the vector"""
+        return awkward.zip(
+            {
+                "x": - self.x,
+                "y": - self.y,
+                "z": - self.z,
+                "t": - self.t
+            },
+            with_name="LorentzVector",
+        )
+
+    @property
+    def pvec(self):
+        """The `x`, `y` and `z` compontents as a `ThreeVector`"""
+        return awkward.zip(
+            {
+                "x": self.x,
+                "y": self.y,
+                "z": self.z
+            },
+            with_name="ThreeVector"
+        )
+
+    @property
+    def boostvec(self):
```

Comment:
If `t <= rho` (or equivalently `mass <= 0`) do we want to make the output NaN since there is no physical way to get to that reference frame?

Reply:
I think this would make sense, but would you then also return NaN for zero vectors (there we have `0 == t == rho`)?

Reply:
As `t` approaches `rho`, `boostvec` approaches the unit vector (i.e. speed of light), so I think it would be consistent if we returned `self.pvec * numpy.minimum(1 / self.rho, 1 / self.t)`.

Reply:
I totally ignored your zero vector comment! Let me try again: `self.pvec * awkward.where(self.rho == 0, 0, numpy.minimum(1 / self.rho, 1 / self.t))`

Reply:
Well - unphysical lorentz transformations still belong to SO(1,3), instead of SO+(1,3). So it's still a lorentz xform but it cannot be described by a boost and a rotation. 

It may be better, and more correct, to simply warn that a boost was calculated outside of the restricted lorentz group. Then also to protect against zeros in some reasonable way.

This may also suggest returning the the boost tensor rather than just the vec... Or extending what we do to include that notion. It then becomes much easier to impose notions of physicality since it's pretty easy to check the lorentz xform tensor for what type it is.

Reply:
I made it depend on `|t| <= rho`. It not returning a normal boost vector even if `|t| >> rho` sounds like a possible pitfall to me and one can easily get into this situation by `(-lorentzvec).boostvec`, which works fine in uproot_methods and ROOT. 
If the user tries to boost with a vector coming from `|t| <= rho` there will be a warning in the `boost` function.

Reply:
Not to wax too long on this but based on https://en.wikipedia.org/wiki/Lorentz_group#Connected_components I think actually improper transforms are outside SO(1, 3). But also |boostvec| > 1 would give complex-valued output in the current `boost` implementation so I'm not sure it makes any sense even as a parity-reversing transform.